Std/Tactic/Basic.lean

```diff
@@ -162,3 +162,27 @@ example (P : (Nat → Nat) → Prop) : P (fun n => n - n) := by
 -/
 macro (name := Conv.equals) "equals " t:term " => " tac:tacticSeq : conv =>
   `(conv| tactic => show (_ = $t); next => $tac)
+
+/-- `lemma` is not supported, please use `theorem` or import `Mathlib.Tactic.Basic` -/
+syntax (name := lemma) declModifiers
+  group("lemma " declId ppIndent(declSig) declVal Parser.Command.terminationSuffix) : command
+
+/-- Elaborator for the `lemma` command, with a boolean flag to control
+whether the command emits a warning and code action
+instructing the user to use `theorem` instead.-/
+def elabLemma' (allow : Bool) : CommandElab := fun stx => do
+  if !allow then
+    let lemmaStx := stx[1][0]
+    Elab.Command.liftTermElabM <|
+      Std.Tactic.TryThis.addSuggestion lemmaStx { suggestion := "theorem" }
+    logWarningAt lemmaStx
+      "`lemma` is not supported, please use `theorem` or import `Mathlib.Tactic.Basic`"
```

Comment:
Same but more important since this is not just documentation.

Reply:
```suggestion
      "`lemma` is not supported, please use `theorem` instead.\n{""
      }note: Mathlib defines a `lemma` command, did you mean to `import Mathlib.Tactic.Basic`?"
```